### ISSUE TITLE
comment about rule 4: "Document within the dockerfile"

### ISSUE THREAD

**Issue by sdettmer** [nuest/ten-simple-rules-dockerfiles#99]
comment about rule 4: "Document within the dockerfile"

 Whether to documentation is inside the `Dockerfile`, a `README.txt` or a formal document going through processes and ending up as signed PDF does not matter for reproducibility I think.

What is important is, that **helpful** documentation exists, I think.

**Comment by vsoch** [comment]
Indeed, although if the documentation is outside the Dockerfile (and then lost) it's not very useful. Given there are projects that just grab/save Dockerfile and not surrounding context, I think Dockerfile comments are fairly important, and external to that a bonus. So they both are important.

**Comment by sdettmer** [comment]
@vsoch Thank you for your quick reply.

If a project documents everything in mark down files, possibly the documentation of the Dockerfile should be there as well. Also if the Dockerfile is generated automatically, it makes not much sense to have documentation there.

I see another risk that higher concepts are forgotten when documenting in Dockerfile instead of specification (in projects that have such). Often people document in Dockerfile (or other files) only "what happens inside", but not why it is needed, where the outputs go to and so on.

Of course you can always use it wrongly, such as copying a Dockerfile without its README.md :)

Of course I agree that documentation is important, but creating a rule to allow to "just grab/save Dockerfile and not surrounding context" does not sound suited to me.

**Comment by vsoch** [comment]
Yes, but you are missing the case where an automated tool has scraped a bunch of repos to find the Dockerfile and save them separately, without context (I've done this several times).
